spec/mailers/producer_mailer_spec.rb

```diff
@@ -128,6 +128,33 @@
     expect(mail.body.encoded).to include(p1.name)
   end
 
+  context 'when flag preferred_show_customer_names_to_suppliers is true' do
```

Comment:
We could add a single test for the case where it's not true, to ensure that the names are not displayed in that case.

Reply:
@andrewpbrett Made required changes and did a force push.